var/spack/repos/builtin/packages/opencv/package.py

```diff
@@ -2,399 +2,1069 @@
+    conflicts("+android_mediandk", when="platform=cray", msg="Android only")
+    conflicts("+android_native_camera", when="platform=darwin", msg="Android only")
+    conflicts("+android_native_camera", when="platform=linux", msg="Android only")
+    conflicts("+android_native_camera", when="platform=cray", msg="Android only")
+    conflicts("+avfoundation", when="platform=linux", msg="iOS/macOS only")
+    conflicts("+avfoundation", when="platform=cray", msg="iOS/macOS only")
+    conflicts("+cap_ios", when="platform=darwin", msg="iOS only")
+    conflicts("+cap_ios", when="platform=linux", msg="iOS only")
+    conflicts("+cap_ios", when="platform=cray", msg="iOS only")
+    conflicts("+carotene", when="target=x86:", msg="ARM/AARCH64 only")
+    conflicts("+carotene", when="target=x86_64:", msg="ARM/AARCH64 only")
+    conflicts("+cpufeatures", when="platform=darwin", msg="Android only")
+    conflicts("+cpufeatures", when="platform=linux", msg="Android only")
+    conflicts("+cpufeatures", when="platform=cray", msg="Android only")
+    conflicts("+cublas", when="~cuda")
+    conflicts("+cudnn", when="~cuda")
+    conflicts("+cufft", when="~cuda")
+    conflicts("+directx", when="platform=darwin", msg="Windows only")
+    conflicts("+directx", when="platform=linux", msg="Windows only")
+    conflicts("+directx", when="platform=cray", msg="Windows only")
+    conflicts("+dshow", when="platform=darwin", msg="Windows only")
+    conflicts("+dshow", when="platform=linux", msg="Windows only")
+    conflicts("+dshow", when="platform=cray", msg="Windows only")
+    conflicts("+gtk", when="platform=darwin", msg="Linux only")
+    conflicts("+ipp", when="target=aarch64:", msg="x86 or x86_64 only")
+    conflicts("+jasper", when="+openjpeg")
+    conflicts("+msmf", when="platform=darwin", msg="Windows only")
+    conflicts("+msmf", when="platform=linux", msg="Windows only")
+    conflicts("+msmf", when="platform=cray", msg="Windows only")
+    conflicts("+msmf_dxva", when="platform=darwin", msg="Windows only")
+    conflicts("+msmf_dxva", when="platform=linux", msg="Windows only")
+    conflicts("+msmf_dxva", when="platform=cray", msg="Windows only")
+    conflicts("+opencl_d3d11_nv", when="platform=darwin", msg="Windows only")
+    conflicts("+opencl_d3d11_nv", when="platform=linux", msg="Windows only")
+    conflicts("+opencl_d3d11_nv", when="platform=cray", msg="Windows only")
+    conflicts("+opengl", when="~qt")
+    conflicts("+tengine", when="platform=darwin", msg="Linux only")
+    conflicts("+tengine", when="target=x86:", msg="ARM/AARCH64 only")
+    conflicts("+tengine", when="target=x86_64:", msg="ARM/AARCH64 only")
+    conflicts("+v4l", when="platform=darwin", msg="Linux only")
+    conflicts("+win32ui", when="platform=darwin", msg="Windows only")
+    conflicts("+win32ui", when="platform=linux", msg="Windows only")
+    conflicts("+win32ui", when="platform=cray", msg="Windows only")
```

Comment:
We can now make the variant conditional on the OS, although that might be difficult to do with the for-loop.